### PR TITLE
change shell script relative path to ${PODS_ROOT} instead of ${SRCROOT}/Pods

### PR DESCRIPTION
**Pull request by Whirlwind**
Keep all script path is relative the Pods folder, rather than `SRCROOT`. So we could use the symlink of `Pods` folder.

This is a separate PR for #7781.

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* change shell script relative path to ${PODS_ROOT} instead of ${SRCROOT}/Pods  
  [Whirlwind](https://github.com/Whirlwind)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/target/aggregate_target.rb [hunk 1]
```diff
@@ -285,38 +291,38 @@ def podfile_dir_relative_path
     #         the user project.
     #
     def xcconfig_relative_path(config_name)
-      relative_to_srcroot(xcconfig_path(config_name)).to_s
```
Comment (segiddins) [inline lib/cocoapods/target/aggregate_target.rb]: can you please keep this as a call to `relative_to_srcroot` ?

Comment (Whirlwind) [reply lib/cocoapods/target/aggregate_target.rb]: I want to remove the all method about the srcroot. All path should not relative the srcroot.

Comment (segiddins) [reply lib/cocoapods/target/aggregate_target.rb]: but this directory is still relative to the SRCROOT

Comment (Whirlwind) [reply lib/cocoapods/target/aggregate_target.rb]: This is a separate PR for #7781.

The #7781 will cause the directory may be relative the SRCROOT.

Comment (amorde) [reply lib/cocoapods/target/aggregate_target.rb]: You could potentially wait until this PR is merged, and then update #7781 to include this change

Comment (Whirlwind) [reply lib/cocoapods/target/aggregate_target.rb]: @amorde I separate the #7781 to this PR and #7879 , and closed #7781 .

**Comment by dnkoutso** [comment]
Not 100% certain but this may have broken integration specs.

**Comment by dnkoutso** [comment]
Seeing diffs:
```diff
+++ b/install_static_swift_modules/after/Pods/Headers/Public/ObjCPod/ObjCPod-library-iOS.modulemap
@@ -1 +1 @@
-/Users/segiddins/Development/Rainforest/CocoaPods/tmp/install_static_swift_modules/transformed/Pods/Target Support Files/ObjCPod-library-iOS/ObjCPod-library-iOS.modulemap
\ No newline at end of file
+/Users/dimitris/Development/ios/cocoapods-mine/tmp/install_static_swift_modules/transformed/Pods/Target Support Files/ObjCPod-library-iOS/ObjCPod-library-iOS.modulemap
```

**Comment by Whirlwind** [comment]
Which branch? I pass all tests on master branch two days  ago.

**Comment by dnkoutso** [comment]
@Whirlwind https://github.com/CocoaPods/cocoapods-integration-specs/pull/178

**Comment by dnkoutso** [comment]
It might not be this PR. Not 100% sure yet.

**Pull request merged**